analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,394 @@
+#     # Gather the data.frame to have columns and values in the format of
+#     # our status call, the fraction of total calls that status call is
+#     # responsible for, and the dominant status call made based on the
+#     # fraction value
+#     gather(status, fraction, -dominant_status) %>%
+#     # Plot our focal status values on the x-axis and the fraction values
+#     # on the y-axis
+#     ggplot(ggplot2::aes(
+#       x = dominant_status,
+#       y = fraction
+#     )) +
+#     theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
+#     geom_jitter(alpha = .2)
+# }
+```
+
+
+### Files and directories
+
+```{r}
+results_dir <- "results"
+
+# Define a logical object for running in CI
+running_in_ci <- params$is_ci
+```
+
+### Read in files
+
+Read in cytoband status file and format it for what we will need in this notebook. 
+
+```{r}
+# Read in the file with consensus CN status data and the UCSC cytoband data --
+# generated in `03-add-cytoband-status-consensus.Rmd`
+consensus_seg_cytoband_status_df <-
+  read_tsv(file.path("results", "consensus_seg_with_ucsc_cytoband_status.tsv.gz")) %>%
+  # Need this to not have `chr`
+  mutate(
+    chr = gsub("chr", "", chr),
+    cytoband = paste0(chr, cytoband)
+  ) %>%
+  select(
+    chromosome_arm,
+    # Distinguish this dominant status that is based on cytobands, from the status
+    dominant_cytoband_status = dominant_status,
+    cytoband,
+    Kids_First_Biospecimen_ID,
+    band_length,
+    gain_fraction,
+    loss_fraction,
+    callable_fraction
+  )
+```
+
+Read in the gene-level data. 
+
+```{r}
+# Read in the annotated gene level CN file
+consensus_seg_autosomes_df <-
+  read_tsv(file.path(results_dir, "consensus_seg_annotated_cn_autosomes.tsv.gz"))
+```
+
+## Define most focal units
+
+### Determine chromosome arm status
+
+```{r}
+# Use our custom function to make the arm status calls
+arm_status_count <- status_majority_caller(
+  consensus_seg_cytoband_status_df,
+  "chromosome_arm",
+  status_threshold
+)
+
+# Display table
+arm_status_count %>%
+  group_by(dominant_status)
+```
+
+These are the chromosome arms that have not been defined as gain or loss -- we want to define their cytoband/gene-level status
+
+```{r}
+# Let's get a vector of the neutral arms
+neutral_arms <- arm_status_count %>%
+  filter(dominant_status == "neutral") %>%
+  dplyr::pull("chromosome_arm")
+```
+
+### Determine cytoband status
+
+We want to include cytoband and gene-level calls for chromosome arms that have not been defined as a gain or loss to make the cytoband-level majority calls.
+
+```{r}
+# Now count the cytoband level calls (for each status call) and define
+# the cytoband as that status if more than 90% of the total counts are
+# for that particular status
+cytoband_status_count <- consensus_seg_cytoband_status_df %>%
+  # Filter the annotated CN data to include only neutral chromosome arms
+  filter(chromosome_arm %in% neutral_arms) %>%
```

Comment:
I realized that this may actually miss some important calls, like when a deletion occurs within an otherwise duplicated region. 

So think the approach here might have to be a bit more nuanced than just filtering on arms. I think we want to join to the arm df, then filter on the merged data, with an extra caveat...

Below is a suggestion, but note that I am not necessarily using the correct column names, so this will not work as is.

```suggestion
cytoband_status_count <- consensus_seg_cytoband_status_df %>%
  left_join(arm_status,  by = c("Kids_First_Biospecimen_ID", "chromosome_arm")
  # Filter the annotated CN data to include only neutral chromosome arms and disagreements
  filter(chromosome_arm %in% c("neutral", "uncallable", "unstable") |
         (dominant_cytoband_status != dominant_arm_status & # bands that disagree with arm, but are not neutral (or uncallable)
         !(dominant_cytoband_status %in% c("neutral", "uncallable", "unstable"))) %>%
```